clients/wallet/src/resubmissions.rs

```diff
@@ -0,0 +1,795 @@
+					.await;
+			}
+		});
+
+		// return the list of tasks, to monitor the resubmission
+		failed_tasks
+	}
+
+	#[doc(hidden)]
+	/// Submits transactions found in the wallet's cache to Stellar.
+	///
+	/// # Arguments
+	/// * `failed_tasks` - a list of receivers that will receive a `TransactionEnvelope`
+	/// if resubmission failed
+	/// * `auto_check` - creates another thread that does the checking for failed resubmissions
+	async fn _resubmit_transactions_from_cache(&self, failed_tasks: FailedTasks, auto_check: bool) {
+		let _ = self.transaction_submission_lock.lock().await;
+
+		//  Log those with errors.
+		let envelopes = match self.get_tx_envelopes_from_cache() {
+			Ok((envs, errors)) => {
+				if !errors.is_empty(){
+					tracing::warn!(
+						"_resubmit_transactions_from_cache(): errors from cache: {errors:?}"
+					);
+				}
+				envs
+			},
+			Err(errors) => {
+				tracing::warn!(
+					"_resubmit_transactions_from_cache(): errors from cache: {errors:?}"
+				);
+				return
+			},
+		};
+
+		// to prevent `error[E0434]: can't capture dynamic environment in a fn item`,
+		// use a closure instead
+		let submit =
+			|envelope: TransactionEnvelope| async { self.submit_transaction(envelope).await };
+
+		// there's nothing to resubmit
+		if envelopes.is_empty() {
+			return
+		}
+		tracing::info!("_resubmit_transactions_from_cache(): resubmitting {:?} envelopes in cache...", envelopes.len());
+
+		let mut error_collector = vec![];
+		// loop through the envelopes and resubmit each one
+		for envelope in envelopes {
+			if let Err(e) = submit(envelope.clone()).await {
+				tracing::debug!("_resubmit_transactions_from_cache(): encountered error: {e:?}");
+				// save the kind of error and the envelope that failed
+				error_collector.push((e, envelope));
+			}
+		}
+
+		// a few errors happened and must be handled.
+		if !error_collector.is_empty() {
+			// clone self, to use in another thread
+			let me = Arc::new(self.clone());
+			let failed_tasks_clone = failed_tasks.clone();
+			tokio::spawn(async move {
+				me.handle_errors(error_collector, failed_tasks_clone).await;
+			});
+		}
+
+		// auto removes a receiver from the list, if a running task has finished
+		let me = Arc::new(self.clone());
+		if auto_check {
+			auto_check_failed_tasks(failed_tasks.clone(), me).await;
+		}
+	}
+
+	#[doc(hidden)]
+	/// Handle all errors
+	///
+	/// # Arguments
+	/// * `errors` - a list of a tuple containing the error and its transaction envelope
+	/// * `failed_tasks` - a list of receivers that will receive a `TransactionEnvelope`
+	async fn handle_errors(
+		&self,
+		#[allow(unused_mut)]
+		mut errors: Vec<(Error, TransactionEnvelope)>,
+		failed_tasks: FailedTasks,
+	) {
+		while let Some((error, env)) = errors.pop() {
+			// pause process for 20 minutes
+			#[cfg(not(test))]
+			pause_process_in_secs(1200).await;
+
+			// for testing purpose, set to 5 seconds
+			#[cfg(test)]
+			pause_process_in_secs(5).await;
```

Comment:
Why do we need to pause it? We are already pausing in the `loop` of the outer function, no?

Reply:
removed